modules/tinymce/src/themes/silver/main/ts/ui/menus/menu/MenuStructures.ts

```diff
@@ -90,28 +96,122 @@ const preprocessCollection = (items: ItemTypes.ItemSpec[], isSeparator: (a: Item
+  columns: number | 'auto',
+  initItems: ItemTypes.ItemSpec[],
+  onItem: (i: ItemTypes.ItemSpec) => ItemTypes.ItemSpec
+) => {
+  return AlloyMenu.parts.items({
+    preprocess: (rawItems: ItemTypes.ItemSpec[]) => {
+      // Add any information to the items that is required. For example
+      // when the items are results in a searchable menu, we need them to have
+      // an ID that can be referenced by aria-activedescendant
+      const enrichedItems = Arr.map(rawItems, onItem);
+      if (columns !== 'auto' && columns > 1) {
+        return chunk<AlloySpec>({
+          tag: 'div',
+          classes: [ 'tox-collection__group' ]
+        }, columns)(enrichedItems);
+      } else {
+        return preprocessCollection(enrichedItems, (_item, i) => initItems[i].type === 'separator');
+      }
+    }
+  });
+};
+
 const forCollection = (columns: number | 'auto', initItems: ItemTypes.ItemSpec[], _hasIcons: boolean = true): StructureSpec => ({
   dom: {
     tag: 'div',
-    classes: [ 'tox-menu', 'tox-collection' ].concat(columns === 1 ? [ 'tox-collection--list' ] : [ 'tox-collection--grid' ])
+    classes: [ 'tox-menu', 'tox-collection' ].concat(
+      columns === 1 ? [ 'tox-collection--list' ] : [ 'tox-collection--grid' ]
+    )
   },
   components: [
-    // TODO: Clean up code and test atomically
-    AlloyMenu.parts.items({
-      preprocess: (items: ItemTypes.ItemSpec[]) => {
-        if (columns !== 'auto' && columns > 1) {
-          return chunk<AlloySpec>({
-            tag: 'div',
-            classes: [ 'tox-collection__group' ]
-          }, columns)(items);
-        } else {
-          return preprocessCollection(items, (_item, i) => initItems[i].type === 'separator');
-        }
-      }
-    })
+    // We don't need to add IDs for each item because there are no
+    // aria relationships we need to maintain
+    insertItemsPlaceholder(columns, initItems, Fun.identity)
   ]
 });
 
+const forCollectionWithSearchResults = (columns: number | 'auto', initItems: ItemTypes.ItemSpec[], _hasIcons: boolean = true): StructureSpec => {
+  // A collection with results is exactly like a collection, except it also has
+  // an ID and class on its outer div to allow for aria-controls relationships, and ids
+  // on its items.
+
+  // This connects the search bar with the list box.
+  const ariaControlsSearchResults = Id.generate('aria-controls-search-results');
+
+  return {
+    dom: {
+      tag: 'div',
+      classes: [ 'tox-menu', 'tox-collection', searchResultsClass ].concat(
+        columns === 1 ? [ 'tox-collection--list' ] : [ 'tox-collection--grid' ]
+      ),
+      attributes: {
+        id: ariaControlsSearchResults
+      }
+    },
+    components: [
+      // For each item, it needs to have an ID, so that we can refer to it
+      // by the aria-activedescendant attribute
+      insertItemsPlaceholder(columns, initItems, augmentWithAria)
+    ]
+  };
+};
+
+// Does a searchable menu *really* support columns !== 1 ?
```

Comment:
Does this need to be tested? I could see a use case like search for icons or what ever.

Reply:
We'd need design specs, I think. Does the search field span all the columns etc.